src/herder/TxSetFrame.cpp

```diff
@@ -229,7 +229,8 @@ TxSetFrame::surgePricingFilter(Application& app)
 
         auto actTxQueueMap = buildAccountTxQueues();
 
-        SurgeCompare const surge(header);
+        auto headerCopy = header.current();
+        SurgeCompare const surge(headerCopy);
```

Comment:
Because in general making copies in leaf functions is bad as at best it can cause performance issues (like if I need to call a different function from `surgePricingFilter`, I would share the same `headerCopy`)

Reply:
This argument sounds like a premature optimization to me. Let's start by writing code that's easy to use correctly (indeed, this interface was already used incorrectly in an earlier version of this PR, so it is not easy to use correctly today) and if we see a performance problem later then we can correct it.